windows/gce-gpu-monitoring-cuda.ps1

```diff
@@ -0,0 +1,227 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https=//www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ 
+<#
+This script can be used for monitoring  GPU utilization on Google Compute Engine
+instances with GPUs attached. It requires the nvidia-smi tool to be properly
+installed on the system (https=//developer.nvidia.com/nvidia-system-management-interface).
+
+This script assumes it is being executed on GCE instance and that it has access
+to the metadata server (https=//cloud.google.com/compute/docs/storing-retrieving-metadata).
+#>
+ 
+ Param(
+    #configure the interval, default is 5 and configure bigger than 5 to avoid throttling
+    $interval = 10,
+
+    #cuda is not supported yet
+    [switch]$collect_each_core = $false
+)
+
+$nvidia_smi_path = "C:\Program Files\NVIDIA Corporation\NVSMI\nvidia-smi.exe"
+$metadata_server = "http://metadata/computeMetadata/v1/instance/"
+$access_token_command = 'gcloud auth application-default print-access-token'
+
+$gpu_metrics = @{
+    'utilization.gpu'= 'instance/gpu/utilization';
+    'utilization.memory'= 'instance/gpu/memory_utilization';
+    'memory.total'= 'instance/gpu/memory_total';
+    'memory.used'= 'instance/gpu/memory_used';
+    'memory.free'= 'instance/gpu/memory_free';
+    'temperature.gpu'= 'instance/gpu/temperature';
+}
+
+
+function get_metrics{
+    <#
+    Calls the nvidia-smi tool to retrieve usage metrics about the
+    attached GPUs.
+    :return:
+    Dictionary that maps (gpu_type, gpu_bus_id) to metric values.
+    #>
+    
+    $query_params = 'gpu_name,gpu_bus_id,'+(($gpu_metrics.Keys|sort-object) -join ",")
+
+    $result = & $nvidia_smi_path --query-gpu=$query_params --format=csv,nounits,noheader
+
+    $metric_name = $query_params -split ","
+
+    $final = @{}
+    foreach($r in $result -split "\r?\n"){
+        $r = ($r -split ",").trim()
+        for( $i =2;$i -lt $r.length;$i++){
+            $final[$r[0]+","+$r[1]]+=@{$metric_name[$i]=$r[$i]}
+        }
+    }
+
+    return $final
+    
+}
+
+function get_timeseries_entry{
```

Comment:
Suggestion: update the function name to `ConvertTo-TimeSeriesEntry` for better aligning with PowerShell best practices.

Reply:
I updated the name as advised. Thanks.